src/app/task/task.component.ts

```diff
@@ -0,0 +1,14 @@
+import { Component, Input } from '@angular/core';
+import { Task } from '../models/task';
+import { NgIf } from '@angular/common';
+
+@Component({
+  selector: 'app-task',
+  standalone: true,
+  imports: [NgIf],
+  templateUrl: './task.component.html',
+  styleUrl: './task.component.scss'
+})
+export class TaskComponent {
+  @Input() task?: Task;
```

Comment:
this is fine, you can also define it like task: Task | null = null;

and in the tempalte use

 @if (task) {
// do stuff 
 }
@else {
// incomplete stuff
}

This will get rid of any uncertainties that a task is not present, but if you want you can leave it as is. I would still add an ngIf check if there is a task, to get rid of the task?.title etc